scintillometry/io/psrfits/core.py

```diff
@@ -160,21 +225,29 @@ class PSRFITSWriter:
     ----------
     filename : str
         Output file name.
-    ih : input file handle
-        The file handle for input data.
     hdus: str, optional
 
 
-
     Notes
     -----
     Currently it only support write the PSRFITS primary HDU and Subint HDU.
     """
 
-    def __init__(self, filename, ih, hdus=None):
-        self.filename = filename
-        self.ih = ih
-        self.hdus = hdus
+    def __init__(self, filename, hdu):
+        self._filename = filename
+        self.hdu = hdu
+
+    def verify(self):
+        # Verify if the input hdus are read to write out.
+        pass
+
+    def write(self, filename=None):
```

Comment:
This should write samples into the HDU, and, once HDU is full, write to disk.

Reply:
If we give HDU shapes, how do we decide if one HDU is full or not?

Reply:
Fixed, Please check.